adapters/repos/db/vector/hnsw/distancer/dot_product_amd64_test.go

```diff
@@ -292,7 +292,7 @@ func testDotProductFloatByteFixedValue(t *testing.T, size uint, dotFn func(x []f
 
 func testDotProductFloatByteRandomValue(t *testing.T, size uint, dotFn func(x []float32, y []byte) float32) {
 	r := getRandomSeed()
```

Comment:
suggestion: idk if reproducibility is an issue here (i.e. can the test pass with seed=17 but fail with seed=31?), but if it is we could include the seed in the failure message and allow settings a seed from command line:

```
go test . -- -seed=31
```